src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_ISymbolInitializer.vb

```diff
@@ -458,5 +458,1362 @@ IFieldInitializerOperation (Field: C.s1 As System.Byte) (OperationKind.FieldInit
+              Instance Receiver: 
+                null
+              Arguments(0)
+
+    Jump if True (Regular) to Block[B3]
+        IIsNullOperation (OperationKind.IsNull, Type: System.Boolean, IsInvalid, IsImplicit) (Syntax: 'M()')
+          Operand: 
+            IFlowCaptureReferenceOperation: 0 (OperationKind.FlowCaptureReference, Type: System.Nullable(Of System.Int32), IsInvalid, IsImplicit) (Syntax: 'M()')
+
+    Next (Regular) Block[B2]
+Block[B2] - Block
+    Predecessors: [B1]
+    Statements (1)
+        IFlowCaptureOperation: 1 (OperationKind.FlowCapture, Type: null, IsInvalid, IsImplicit) (Syntax: 'M()')
+          Value: 
+            IInvocationOperation ( Function System.Nullable(Of System.Int32).GetValueOrDefault() As System.Int32) (OperationKind.Invocation, Type: System.Int32, IsInvalid, IsImplicit) (Syntax: 'M()')
+              Instance Receiver: 
+                IFlowCaptureReferenceOperation: 0 (OperationKind.FlowCaptureReference, Type: System.Nullable(Of System.Int32), IsInvalid, IsImplicit) (Syntax: 'M()')
+              Arguments(0)
+
+    Next (Regular) Block[B4]
+Block[B3] - Block
+    Predecessors: [B1]
+    Statements (1)
+        IFlowCaptureOperation: 1 (OperationKind.FlowCapture, Type: null, IsInvalid, IsImplicit) (Syntax: 'M2')
+          Value: 
+            IInvocationOperation (Function C.M2() As System.Int32) (OperationKind.Invocation, Type: System.Int32, IsInvalid) (Syntax: 'M2')
+              Instance Receiver: 
+                null
+              Arguments(0)
+
+    Next (Regular) Block[B4]
+Block[B4] - Block
+    Predecessors: [B2] [B3]
+    Statements (1)
+        ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Int32, IsInvalid, IsImplicit) (Syntax: '= If(M(), M2)')
+          Left: 
+            IFieldReferenceOperation: C.s1 As System.Int32 (Static) (OperationKind.FieldReference, Type: System.Int32, IsInvalid, IsImplicit) (Syntax: '= If(M(), M2)')
+              Instance Receiver: 
+                null
+          Right: 
+            IFlowCaptureReferenceOperation: 1 (OperationKind.FlowCaptureReference, Type: System.Int32, IsInvalid, IsImplicit) (Syntax: 'If(M(), M2)')
+
+    Next (Regular) Block[B5]
+Block[B5] - Exit
+    Predecessors: [B4]
+    Statements (0)
+]]>.Value
+
+            Dim expectedDiagnostics = <![CDATA[
+BC30059: Constant expression is required.
+    Const s1 As Integer = If(M(), M2)'BIND:"= If(M(), M2)"
+                          ~~~~~~~~~~~
+]]>.Value
+
+            VerifyFlowGraphAndDiagnosticsForTest(Of EqualsValueSyntax)(source, expectedFlowGraph, expectedDiagnostics)
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)>
+        <Fact>
+        Public Sub ControlFlow_AsNewFieldInitializer()
+            Dim source = <![CDATA[
+Class C
+    Private s1, s2 As New C(If(a, b))'BIND:"As New C(If(a, b))"
```

Comment:
>Private s1, s2 As New C(If(a, b)) [](start = 4, length = 33)

Please confirm the order of evaluation by examining IL for this scenario.

Reply:
Verified:

```
  IL_0009:  ldsfld     valuetype [mscorlib]System.Nullable`1<int32> ClassLibrary1.C::a
  IL_000e:  dup
  IL_000f:  stloc.0
  IL_0010:  stloc.1
  IL_0011:  ldloca.s   V_1
  IL_0013:  call       instance bool valuetype [mscorlib]System.Nullable`1<int32>::get_HasValue()
  IL_0018:  brtrue.s   IL_0021
  IL_001a:  ldsfld     int32 ClassLibrary1.C::b
  IL_001f:  br.s       IL_0028
  IL_0021:  ldloca.s   V_0
  IL_0023:  call       instance !0 valuetype [mscorlib]System.Nullable`1<int32>::GetValueOrDefault()
  IL_0028:  newobj     instance void ClassLibrary1.C::.ctor(int32)
  IL_002d:  stfld      class ClassLibrary1.C ClassLibrary1.C::s1
  IL_0032:  ldarg.0
  IL_0033:  ldsfld     valuetype [mscorlib]System.Nullable`1<int32> ClassLibrary1.C::a
  IL_0038:  dup
  IL_0039:  stloc.0
  IL_003a:  stloc.1
  IL_003b:  ldloca.s   V_1
  IL_003d:  call       instance bool valuetype [mscorlib]System.Nullable`1<int32>::get_HasValue()
  IL_0042:  brtrue.s   IL_004b
  IL_0044:  ldsfld     int32 ClassLibrary1.C::b
  IL_0049:  br.s       IL_0052
  IL_004b:  ldloca.s   V_0
  IL_004d:  call       instance !0 valuetype [mscorlib]System.Nullable`1<int32>::GetValueOrDefault()
  IL_0052:  newobj     instance void ClassLibrary1.C::.ctor(int32)
  IL_0057:  stfld      class ClassLibrary1.C ClassLibrary1.C::s2
```

---
In reply to: [187467860](https://github.com/dotnet/roslyn/pull/26763#discussion_r187467860) [](ancestors = 187467860)